src/webSocket.ts

```diff
@@ -0,0 +1,60 @@
+import { EventEmitter } from '@byungi/event-emitter'
+import {
+  AcceptNanoPayment,
+  isAcceptNanoPayment,
+  isCompletedAcceptNanoPayment,
+} from './types'
+import { logger } from './logger'
+
+export const createWebSocketURL = ({
+  baseURL,
+  paymentToken,
+}: {
+  baseURL: string
+  paymentToken: AcceptNanoPayment['token']
+}) => `${baseURL}?token=${paymentToken}`
+
+type AcceptNanoWebSocketEvents = {
+  open: () => void
+  close: () => void
+  error: (error: unknown) => void
+  payment_updated: (payment: AcceptNanoPayment) => void
+  payment_completed: (payment: AcceptNanoPayment) => void
```

Comment:
maybe it's better to call this `payment_verified` and get rid of `completed` keyword altogether. we have both `completed payment` and `verified payment` notions but they're identical lol wtf